docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -195,7 +193,63 @@ Quarkus provides additional extensions which cover other configuration formats a
 
 TIP: It is also possible to create a xref:config-extending-support.adoc#custom-config-source[Custom Config Source].
 
-== Inject
+== ConfigMapping
+
+The recommended way to expose configuration to a Quarkus application is by mapping it with an interface. With
+https://smallrye.io/smallrye-config/Main/config/mappings/[SmallRye Config Mappings], it is possible to
+group multiple configuration properties in a single interface that share the same prefix (or namespace). It supports
+the following set of features:
+
+* Automatic conversion of the configuration type, including List, Set, Map, Optional, and primitive types.
+* Nested Config Mapping groups.
+* Configuration Properties Naming Strategies
+* Integration with Bean Validation
+
+A Config Mapping requires an interface with minimal metadata configuration annotated with
+`@io.smallrye.config.ConfigMapping`:
+
+[source,java]
+----
+@ConfigMapping(prefix = "server")
+public interface Server {
+    String host();
+
+    int port();
+}
+----
+
+The `Server` interface can map configuration properties with the name `server.host` into the `Server.host()`
+method and `server.port` into `Server.port()` method. The configuration property name to look up is constructed from
+the prefix and the method name, separated by a dot (`.`) character.
+
+A `@ConfigMapping` can be injected into any CDI aware bean:
+
+[source,java]
+----
+@ApplicationScoped
+class BusinessBean {
+    @Inject
+    Server server;
+
+    void businessMethod() {
+        String host = server.host();
+    }
+}
+----
+
+For non-CDI components, use the API `io.smallrye.config.SmallRyeConfig#getConfigMapping` to retrieve the config mapping
+instance:
+
+[source,java]
+----
+SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
+Server server = config.getConfigMapping(Server.class);
+----
+
+TIP: Check the xref:config-mappings.adoc[Mapping configuration to objects] guide for advanced usages with
+`@ConfigMapping`.
+
+== ConfigProperty
 
 Quarkus uses https://microprofile.io/project/eclipse/microprofile-config[MicroProfile Config] annotations to inject the
```

Comment:
Maybe: While `@ConfigMapping` is the recommended approach, Quarkus also supports...

Reply:
Sounds good.